Try mapping to schema

ndaelman-hu opened the pull request:
using example https://raw.githubusercontent.com/nomad-coe/electronic-parsers/develop/tests/data/vasp/AgAc_relax/vasprun.xml.relax

Commit:
Try mapping to schema

Commit:
Apply reviewer feedback

Commit:
Correct tag attribute path in jmespath

ladinesa commented on src/nomad_parser_vasp/parsers/xml_parser.py:
```diff
@@ -23,83 +23,88 @@
-    xml=MappingAnnotationModel(path='modeling.generator.i[@name="program"]')
+Simulation.m_annotations = dict(xml=MappingAnnotationModel(path='modeling'))
+Simulation.program.name.m_annotations = dict(
+    xml=MappingAnnotationModel(path='modeling.generator.i[?"@name"="program"]')
```
you can simply use .i[name=program]here

ndaelman-hu replied:
you mean `.i[?"@name"=program]`, right? Why is this faster, though.

ladinesa commented on src/nomad_parser_vasp/parsers/xml_parser.py:
```diff
@@ -23,83 +23,88 @@
-Program.compilation_host.m_annotations = dict(
-    xml=MappingAnnotationModel(path='modeling.generator.i[@name="platform"]')
+Simulation.program.compilation_host.m_annotations = dict(
+    xml=MappingAnnotationModel(path='modeling.generator.i[?"@name"="platform"]')
```
also here. this should also work btw but a little slower since jmespath will do a global search

ladinesa commented on src/nomad_parser_vasp/parsers/xml_parser.py:
```diff
@@ -23,83 +23,88 @@
 KMesh.grid.m_annotations = dict(
-    xml=MappingAnnotationModel(path='modeling.kpoints.generation.v[@name="divisions"]')
+    xml=MappingAnnotationModel(
+        path='modeling.kpoints.generation.v[?"@name"="divisions"]'
```
i suggest using relative path as much as possible

ndaelman-hu replied:
that's why I was asking about the parent. Or can I just leave it out?

ladinesa commented on src/nomad_parser_vasp/parsers/xml_parser.py:
```diff
@@ -23,83 +23,88 @@
+    xml=MappingAnnotationModel(path='calculation.structure.varray[?"@name"="stress"]')
 )
 """
 AtomicCell.lattice_vectors.m_annotations = dict(
```
i think calculation is  repeating xml section so this will not work unless you provide annotatio. to the metainfo section that correpond to calculatio

ndaelman-hu replied:
I can invent my own section, right? Just to reference. Does not have to be of our schema.

ladinesa commented on src/nomad_parser_vasp/parsers/xml_parser.py:
```diff
@@ -23,83 +23,88 @@
 ElectronicEigenvalues.spin_channel.m_annotations = dict(
     xml=MappingAnnotationModel(
-        path='calculation.eigenvalues.array.set.set[@comment="spin 1"]'
+        path='calculation.eigenvalues.array.set.set[?"@comment"="spin 1"]'
```
i think in vasprun the eigenalues and occupations are in the same element so you would need to introduce an operator to manipulate the data. so you would need to parse all set's and these are lists so the jmespath syntax is set[]

ndaelman-hu replied:
Can't I  use the jmespath index? E.g.

`ElectronicEigenvalues.reciprocal_cell.m_annotations.xml + '.r[0]'`

Commit:
- Fix plugin setup
- Distinguish between parser reference and entry point reference
- TODO: fix `m_annotations` overloading

Commit:
Fix annotations

ndaelman committed:
Add back in the annotated output

ndaelman committed:
Migrate schema annotations to `schema_p`

ndaelman-hu committed:
Merge pull request #2 from FAIRmat-NFDI/plugin_setup

Settle on the appropriate annotation format.

ndaelman committed:
Update `Simulation` import in `xml_parser.py`

ndaelman-hu committed:
- Update compability to new `MatchingParserInterface` and `ParserEntryPoint`
- TODO: fix empty extraction

ndaelman-hu committed:
Apply Ruff format

ladinesa commented on src/nomad_parser_vasp/parsers/xml_parser.py:
```diff
@@ -6,19 +6,19 @@
 
 
-class VasprunXMLParser(MatchingParser):
+class VasprunXMLParser(MatchingParserInterface):
```
I suggest you do not inherit it here but in entry point load function you return MatchingParserInterface(lself.dict). There is some lazy loading happing which is not used when like this

ndaelman-hu replied:
~~Out of curiosity, why can `parser_class_name` not be passed as an argument too?
It seems like it should be set as a class variable.~~

Okay, I see why. For some reason, it's out filtered when passed to the entry point initializer.

ladinesa commented on src/nomad_parser_vasp/parsers/__init__.py:
```diff
@@ -8,7 +8,7 @@ class VasprunXMLEntryPoint(ParserEntryPoint):
         from nomad_parser_vasp.parsers.xml_parser import VasprunXMLParser
 
-        return VasprunXMLParser(**self.dict())
+        return VasprunXMLParser('nomad_parser_vasp/parser/xml_parser.py/VasprunXMLParser', **self.dict())
```
the parser_class_name should have a module format: nomad_parser_vasp.VasprunXMLParser. Return MatchingParserInterface here

ndaelman-hu replied:
I saw this from Joseph too. I'll change it, but this one works as well ;)

ndaelman-hu committed:
Update `EntryPoint` setup

ndaelman-hu committed:
Clarify `parser_class_name` role

ndaelman-hu committed:
Modify JMesPath quotation symbols

ndaelman-hu committed:
Modify JMesPath quotation symbols

ladinesa committed:
Partial fix

ndaelman-hu committed:
Merge pull request #7 from ladinesa/first_version-fix

Example fix for the jmespath path specification

ndaelman-hu committed:
Move `mix_alpha` outside parser class
